File: common/src/main/java/dev/niostone/interstellar/config/StarSystemConfig.java
package dev.niostone.interstellar.config;

import java.util.List;
import java.util.ArrayList;

import me.shedaniel.autoconfig.ConfigData;
import me.shedaniel.autoconfig.annotation.Config;

@Config(name = "system")
public class StarSystemConfig implements ConfigData {
    public int id;
    public String name;
    public List<Integer> planets;

    public StarSystemConfig(String name) {
        this.name = name;
        this.planets = new ArrayList<>();
    }
}

File: common/src/main/kotlin/dev/niostone/interstellar/impl/config/StarSystemConfig.kt
package dev.niostone.interstellar.impl.config

import dev.niostone.interstellar.impl.Interstellar
import me.shedaniel.autoconfig.ConfigData
import me.shedaniel.autoconfig.annotation.Config

@Config(name = "system")
class StarSystemConfig: ConfigData {
    var id: Int = -1
    var name: String
    var planets: MutableList<Int>

    constructor() : this(Interstellar.MOD_ID + ":null")

    constructor(name: String) {
        this.name = name;
        this.planets = mutableListOf()
    }
}
